### DISCUSSION TITLE
Best way of rendering content of editor?

### DISCUSSION THREAD

**Discussion by michalvadak** [facebook/lexical#2455] [Q&A] [answered]
What is the best approach of just showing the content of lexical that was written in administration to be shown in the client zone without the ability to edit it?

1. Save the JSON to DB and then on the frontend pass the JSON to the lexical editor with `readOnly: true`?
2. Save the JSON to DB and then on the frontend pass the JSON to the custom component that will render the content?
3. Searialize to HTML before sending do DB?

**Comment by thegreatercurve** [comment, marked as answer]
There's really no definitive answer. It's whatever is best for your specific use case. 

The Lexical `readOnly` mode on the editor is the easiest way to show serialized HTML _or_ JSON editor state. 

If client-side performance more of a concern though, you can build a separate "read-only" Lexical Editor which strips out any unnecessary plugins which are used in editing (e.g. `OnChangePlugin`).

For your specific use case, if you're display HTML which is saved in an admin panel elsewhere, you could just remove Lexical altogether on the client app and just render the saved HTML, though you would need to re-apply any custom styles.

**Reply by mubshrx** [reply]
Is https://threads.net using the Lexical in readonly mode approach?

**Reply by potofpie** [reply]
I'd like to know the answer to this puppy 🐶👆

**Reply by elbotho** [reply]
> Is https://threads.net/ using the Lexical in readonly mode approach?

Lexical dev tools don't find an instance.

**Reply by code-mastero** [reply]
@mhm13dev What you end up with ?

**Reply by mubshrx** [reply]
@code-mastero for what?

**Reply by pranavmappoli** [reply]
@mhm13dev ,you have used  `Lexical readOnly mode ` or something else ?

**Reply by mubshrx** [reply]
> you have used Lexical readOnly mode or something else

I stored the json state in db and then rendered the nodes from json on client side.

**Reply by pranavmappoli** [reply]
@mhm13dev how you rendered in client side ? same with lexical editor ?

**Reply by mubshrx** [reply]
> @mhm13dev how you rendered in client side ? same with lexical editor ?
> 

No, I used custom components to render the contents based on node types.

I don't have the code example handy now.

I will share once I have it.

**Comment by vannsoklay** [comment]
how to render json lexical from db to html on solidjs.
1. [json laxical] -> db
2. query -> [1] -> html 
3. client -> [2] -> read html

**Comment by vadimkantorov** [comment]
TWIMC I made a primitive example achieving lexical editorState JSON -> HTML rendering: https://github.com/vadimkantorov/lexical-cli-render (as a CLI tool)
